modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx

```diff
@@ -77,11 +88,30 @@ const RequiredInformation = ({
 
 	const isOemOrEnterprise =
 		infoSelectedKey?.licenseEntryType.includes('OEM') ||
-		infoSelectedKey?.licenseEntryType.includes('Enterprise');
+		infoSelectedKey?.licenseEntryType.includes('Enterprise') ||
+		(state.activationKeys.length &&
+			(state?.activationKeys[0].licenseEntryType.includes('oem') ||
+				state?.activationKeys[0].licenseEntryType.includes(
+					'enterprise'
+				)));
+
+	const hasNotPermanentLicenceRenewKey =
```

Comment:
`hasNotPermanent` isnt gramatically correct. what about something like `hasRestrictedPermanentLicenseRenewKey`? since these are the types that are restricted and can't have a permanent key